components/console-backend-service/internal/domain/eventing/trigger/service.go

```diff
@@ -0,0 +1,182 @@
+package trigger
+
+import (
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/domain/eventing/pretty"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlschema"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/resource"
+	notifierResource "github.com/kyma-project/kyma/components/console-backend-service/pkg/resource"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+//go:generate mockery -name=Service -output=automock -outpkg=automock -case=underscore
+//go:generate failery -name=Service -case=underscore -output disabled -outpkg disabled
+type Service interface {
+	Create(trigger *v1alpha1.Trigger) (*v1alpha1.Trigger, error)
+	CreateMany(triggers []*v1alpha1.Trigger) ([]*v1alpha1.Trigger, error)
+	Delete(trigger gqlschema.TriggerMetadataInput) error
+	DeleteMany(triggers []gqlschema.TriggerMetadataInput) error
+	List(namespace string, subscriber *gqlschema.SubscriberInput) ([]*v1alpha1.Trigger, error)
+	CompareSubscribers(fir *gqlschema.SubscriberInput, sec *v1alpha1.Trigger) bool
+	Subscribe(listener notifierResource.Listener)
+	Unsubscribe(listener notifierResource.Listener)
+}
+type triggerService struct {
+	*resource.Service
+	notifier notifierResource.Notifier
+}
+
+func NewService(serviceFactory *resource.ServiceFactory) *triggerService {
+	svc := &triggerService{
+		Service: serviceFactory.ForResource(schema.GroupVersionResource{
+			Group:    v1alpha1.SchemeGroupVersion.Group,
+			Version:  v1alpha1.SchemeGroupVersion.Version,
+			Resource: "triggers",
+		}),
+		notifier: notifierResource.NewNotifier(),
```

Comment:
You must connect notifier to informer. Without this subscriptions won't work.

```go
svc := &triggerService{
		Service: serviceFactory.ForResource(schema.GroupVersionResource{
			Group:    v1alpha1.SchemeGroupVersion.Group,
			Version:  v1alpha1.SchemeGroupVersion.Version,
			Resource: "triggers",
		}),
}

	notifier := notifierResource.NewNotifier()
	svc.Informer.AddEventHandler(notifier)
	svc.notifier = notifier

return svc
```

Also please use indexers like in Rafter -> https://github.com/kyma-project/kyma/blob/master/components/console-backend-service/internal/domain/rafter/asset_service.go#L46